frontend/src/interactionContext.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import apiEndpoints from "./apiEndpoints";
+import axios from 'axios';
+import {toast} from "sonner";
+
+const InteractionContext = createContext();
+
+export const InteractionProvider = ({ children }) => {
+    const [interactionTypes, setInteractionTypes] = useState(() => {
+        return JSON.parse(localStorage.getItem('interactionTypes')) || {};
+    });
+
+    useEffect(() => {
+        const getInteractionTypes = async () => {
+            try {
+                const response = await axios.get(`${apiEndpoints.url}${apiEndpoints.interactions.getTypes}`);
+                const types = response.data;
+                localStorage.setItem('interactionTypes', JSON.stringify(types));
```

Comment:
MINOR:

I do agree with Andrew on this: it's not the best idea to hold these inside local storage. I would probably go for caching response on our API endpoint if I were you. Or maybe some dynamic JS varibale.

BUT this is alright. Leave it be